examples/optimization/demo_3D_wavelets.py

```diff
@@ -0,0 +1,245 @@
+"""
+3D wavelet denoising
+====================================================================================================
+
+This example shows how to use a 3D wavelet denoiser for denoising a 3D image. We first apply a standard soft-thresholding
+wavelet denoiser to a 3D brain MRI volume. We then extend the denoiser objective to a redundant dictionary of wavelet
+bases, which does not admit a closed-form solution. We solve the denoising problem using the Dykstra-like algorithm.
+"""
+
+import deepinv as dinv
+from pathlib import Path
+import numpy as np
+import matplotlib.pyplot as plt
+
+import torch
+import torch.nn as nn
+
+from deepinv.utils.demo import load_url_data
+
+# %%
+# Setup paths for data loading and results.
+# ----------------------------------------------------------------------------------------
+#
+
+BASE_DIR = Path(".")
+ORIGINAL_DATA_DIR = BASE_DIR / "datasets"
+DATA_DIR = BASE_DIR / "measurements"
+RESULTS_DIR = BASE_DIR / "results"
+DEG_DIR = BASE_DIR / "degradations"
+
+
+# %%
+# Load base volume image and denoising operators.
+# ----------------------------------------------------------------------------------------
+# In this example, we use a T1-weighted brain MRI volume from the BrainWeb dataset (subject id 4) and we add
+# Gaussian random noise to it.
```

Comment:
maybe explain the size of the tensor representing a volume?

is it B x C x H x W x P ?

Reply:
Done (the default torch convention is BxCxDxHxW (depth is after C)